src/main/java/org/jitsi/srtp/BaseSrtpCryptoContext.java

```diff
@@ -191,96 +166,61 @@ protected BaseSrtpCryptoContext(
-            {
-                cipherCtr = new SrtpCipherCtrOpenSsl();
-            }
-            else
-            {
-                cipherCtr
-                    = new SrtpCipherCtrJava(Aes.createStreamCipher(encKeyLength));
-            }
-            saltKey = new byte[saltKeyLength];
+            cipher = new SrtpCipherF8(Cipher.getInstance("AES/ECB/NoPadding"));
             break;
-
         case SrtpPolicy.TWOFISHF8_ENCRYPTION:
-            cipherF8 = new SrtpCipherF8(new TwofishEngine());
-            //$FALL-THROUGH$
-
+            cipher = new SrtpCipherF8(Cipher.getInstance("Twofish/ECB/NoPadding"));
+            break;
         case SrtpPolicy.TWOFISH_ENCRYPTION:
-            cipherCtr = new SrtpCipherCtrJava(new SICBlockCipher(new TwofishEngine()));
-            saltKey = new byte[saltKeyLength];
+            cipher = new SrtpCipherCtr(Cipher.getInstance("Twofish/CTR/NoPadding"));
+            break;
+        case SrtpPolicy.NULL_ENCRYPTION:
+        default:
+            cipher = null;
             break;
         }
-        this.cipherCtr = cipherCtr;
-        this.cipherF8 = cipherF8;
-        this.saltKey = saltKey;
 
         Mac mac;
-        byte[] tagStore;
-
         switch (policy.getAuthType())
         {
         case SrtpPolicy.HMACSHA1_AUTHENTICATION:
-            mac = HmacSha1.createMac();
-            tagStore = new byte[mac.getMacSize()];
+            mac = HmacSha1.createMac(parentLogger);
             break;
 
         case SrtpPolicy.SKEIN_AUTHENTICATION:
-            tagStore = new byte[policy.getAuthTagLength()];
-            mac = new SkeinMac(SkeinMac.SKEIN_512, tagStore.length * 8);
+            mac = Mac.getInstance("SkeinMac_512_" + (policy.getAuthTagLength() * 8));
             break;
 
         case SrtpPolicy.NULL_AUTHENTICATION:
         default:
             mac = null;
-            tagStore = null;
             break;
         }
         this.mac = mac;
-        this.tagStore = tagStore;
     }
 
     /**
-     * Authenticates a packet. Calculated authentication tag is returned/stored
-     * in {@link #tagStore}.
+     * Authenticates a packet.
      *
      * @param pkt the RTP packet to be authenticated
      * @param rocIn Roll-Over-Counter
      */
-    synchronized protected void authenticatePacketHmac(ByteArrayBuffer pkt, int rocIn)
+    synchronized protected byte[] authenticatePacketHmac(ByteArrayBuffer pkt, int rocIn)
     {
         mac.update(pkt.getBuffer(), pkt.getOffset(), pkt.getLength());
         rbStore[0] = (byte) (rocIn >> 24);
         rbStore[1] = (byte) (rocIn >> 16);
         rbStore[2] = (byte) (rocIn >> 8);
         rbStore[3] = (byte) rocIn;
         mac.update(rbStore, 0, rbStore.length);
-        mac.doFinal(tagStore, 0);
-    }
-
-    /**
-     * Closes this crypto context. The close functions deletes key data and
-     * performs a cleanup of this crypto context.
-     */
-    synchronized public void close()
-    {
-        /* TODO, clean up ciphers and mac. */
```

Comment:
The goal of this is to erase secret information - I'd rather implement it than lose it.  (Is there any way to do this through the JCE API?)

Reply:
For anything held by the JVM, IMO this is quite pointless. It is free to move memory during GC runs, and I'd bet that old memory regions wouldn't be zeroed. If anything can read your memory, you have other problems.

If you really want to keep it, implement it with `AutoCloseable` so that try-with-resources could be used.